SECP-service/src/main/resources/assets/app/js/chat/chat-controller.js

```diff
@@ -36,20 +36,77 @@ angular.module('SECP')
         }
       });
 
-      $scope.sendMessage = function() {
-         var messageDTO = {
-            groupId: $scope.selectedChat.groupID,
-            senderId: $scope.currentUser.userID,
-            body: $scope.messageInput,
-            reason: "message",
-            timestamp: moment()
-         };
-
-         Socket.send(messageDTO);
-         $scope.messages.push(messageDTO)
-         setLastMessageForContacts(messageDTO.groupId, messageDTO);
-         //clearing the message input in the textarea
-         $scope.messageInput = null;
+      $scope.sendMessage = function(){
+          var promise = $scope.messageContainsFilterWord();
+          promise.then(function(messageContainsFilterWord) {
+              if(!messageContainsFilterWord){
+                  var messageDTO = {
+                      groupId: $scope.selectedChat.groupID,
+                      senderId: $scope.currentUser.userID,
+                      body: $scope.messageInput,
+                      reason: "message",
+                      timestamp: moment()
+                  };
+
+                  Socket.send(messageDTO);
+                  $scope.messages.push(messageDTO)
+                  setLastMessageForContacts(messageDTO.groupId, messageDTO);
+                  //clearing the message input in the textarea
+                  $scope.messageInput = null;
+              }
+          });
+        };
+
+      $scope.messageContainsFilterWord = function() {
+          var deffered = $q.defer();
+          let senderPermission = $scope.currentUser.permission
```

Comment:
if you're gunna go all in on 'let' and 'const', get rid of 'var' lol.

Reply:
Done.